modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1325,6 +1327,30 @@ export class Wallet implements IWallet {
       }
     }
 
+    // Validate EVM keyring params, referenceAddress is required and referenceCoin is optional for EVM keyring
+    if (!_.isUndefined(referenceAddress)) {
+      if (!_.isString(referenceAddress)) {
+        throw new Error('referenceAddress has to be a string');
+      }
+      if (!this.baseCoin.isEVM()) {
+        throw new Error('referenceAddress is only supported for EVM chains');
+      }
+      if (!this.baseCoin.isValidAddress(referenceAddress)) {
+        throw new Error('referenceAddress must be a valid address');
+      }
+      addressParams.referenceAddress = referenceAddress;
+
+      if (!_.isUndefined(referenceCoin)) {
+        if (!_.isString(referenceCoin)) {
+          throw new Error('referenceCoin has to be a string');
+        }
+        addressParams.referenceCoin = referenceCoin;
+      }
+    } else if (!_.isUndefined(referenceCoin)) {
+      // referenceCoin cannot be used without referenceAddress
+      throw new Error('referenceAddress is required when using referenceCoin for EVM keyring');
+    }
+
```

Comment:
Can you instead do a `if(_.isUndefined(referenceCoin) && !_.isUndefined(referenceAddress)) { throw error }`
That's more readable.